pkg/ebpftracer/c/tracee.bpf.c

```diff
@@ -2360,6 +2374,11 @@ CGROUP_SKB_HANDLE_FUNCTION(proto_udp)
 
 CGROUP_SKB_HANDLE_FUNCTION(proto_tcp_dns)
 {
+    // Skip the 2-byte length prefix for dns over tcp.
+    if (net_l7_empty_dns_answer(ctx, md.header_size + 2)) {
```

Comment:
I do not think this is a good idea to be honest. This could hide e.g. recon activity.

Reply:
Will implement it once needed. We do not have it yet and on k8s it's way more tricky.